qiskit/visualization/pass_manager_visualization.py

```diff
@@ -0,0 +1,147 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2019.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+Visualization function for a pass manager. Passes are grouped based on their
+flow controller, and coloured based on the type of pass.
+"""
+import inspect
+from qiskit.transpiler.basepasses import AnalysisPass, TransformationPass
+DEFAULT_STYLE = {AnalysisPass: 'red',
+                 TransformationPass: 'blue'}
+
+
+try:
+    import subprocess
+    print(subprocess.check_output(['dot', '-V']))
+    HAS_GRAPHVIZ = True
+except FileNotFoundError:
+    # this is raised when the dot command cannot be found, which means GraphViz
+    # isn't installed
+    HAS_GRAPHVIZ = False
+
+
+def pass_manager_drawer(pass_manager, filename, style=None):
```

Comment:
This looks good! Are we moving towards using type hinting instead of using the docstrings? I think it is good to keep it all as consistent as possible throughout the codebase.

Reply:
I believe so! E.g. https://github.com/Qiskit/qiskit-terra/pull/2435/files

Reply:
Please do not add types at the moment.  This is under discussion.